tools/runners/Yosys.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+
+from Runner import Runner
+
+
+class Yosys(Runner):
+    def __init__(self):
+        super().__init__("yosys")
+
+    def run_cb(self, tmp_dir, params):
+        scr = os.path.join(tmp_dir, 'scr.ys')
+
+        cmd = ['yosys']
+        cmd.append('-Q')
+        cmd.append('-T')
+        cmd.append('scr.ys')
+
+        inc = ""
+
+        for incdir in params['incdirs']:
+            inc = inc + ' -I' + incdir
+
+        with open(scr, 'w') as f:
+            for svf in params['files']:
+                f.write('read_verilog -sv' + inc + ' ' + svf + '\n')
+
+        proc = subprocess.Popen(cmd, cwd=tmp_dir,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT)
+
+        log, _ = proc.communicate()
+
+        return (log.decode('utf-8'), proc.returncode)
```

Comment:
It seems that all of the runners share the last part (running the sub-process and returning the same stuff). Why is this not some common code if we have a common `Runner.py` anyway?

Reply:
This common part is now inside `BaseRunner` and is shared by almost all runners. `Fake` runner overrides this as it must set the return code based on what is expected by the test case